### ISSUE TITLE
Fix Documentation

### ISSUE THREAD

**Issue by mrocklin** [dask/dask-image#38]
Here are a few notes about the documentation.  

-  Replace stock "Welcome to dask-image’s documentation!" with something like "Image Processing with Dask"
-  Add contents to the main page other than the TOC
-  README has TODO entries and references to cookiecutter
-  Usage file is non-informative, maybe just remove it?
-  Examples in docstrings refer to scipy, maybe they should be reworked or removed?
-  core and imread API docs are empty

**Comment by mrocklin** [comment]
> What other content would you imagine would be useful on the main page?

Ideally the main page gives a brief summary of the capabilities of the project.  It looks like some of this content is in the README.  I would probably drop all of the content of the README into the main page.  I don't think that there is a reason for it to be separate (other than convenience of development).  

I might also include a simple code example designed to entice users to read more.  Maybe something that reads in a bunch of images, and then calls some transformation or processing function within dask-image.  Doesn't necessarily need to be user-runnable or real-world, just something to give users a sense of what they might use this for.

**Comment by mrocklin** [comment]
> With the docstrings was trying to mirror the strategy in Dask with NumPy to keep overhead low, is this confusing from a first read? How would you imagine reworking them? Should we revisit this in Dask as well?

Yeah, totally understood.  I think that this choice is easier to make if there are other examples around explaining how to use things.  As things stand though there aren't.  For young projects I tend to shove all of the documentation into docstrings and have done with it.  This might be a decent place to invest effort.

**Comment by jakirkham** [comment]
Thanks for the follow-ups. This all sounds reasonable. Will address these in subsequent PRs.

**Comment by jakirkham** [comment]
Have given the documentation/examples some more thought and IMHO we really need somewhere to store example data for users to play with and we need some Jupyter Notebooks that follow the progression of some short image processing stories (e.g. [the scikit-image gallery]( http://scikit-image.org/docs/stable/auto_examples/ )). We could even provide users a Binder instance to then play around with this data and our libraries.
